### PR TITLE
#10883 adding const error string DismissPostError to be returned by MessageWillBePosted

### PR DESCRIPTION
**Pull request by carmo-evan**
#### Summary
Adds an exported const string that can be used by plugins to signal a client that the incoming post should be dismissed, instead of the current behavior which just leaves it pending.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/10883

### CODE REVIEW

File: i18n/en.json [hunk 1]
```diff
@@ -7154,4 +7158,4 @@
     "id": "web.incoming_webhook.user.app_error",
     "translation": "Couldn't find the user"
   }
-]
+]
```
Comment (hanzei) [inline i18n/en.json]: There has to be a new line at the end of the file.

File: i18n/en.json [hunk 2]
```diff
@@ -5234,6 +5234,10 @@
     "id": "oauth.gitlab.tos.error",
     "translation": "GitLab's Terms of Service have updated. Please go to gitlab.com to accept them and then try logging into Mattermost again."
   },
+  {
+    "id": "plugin.message_will_be_posted.dismiss_post",
```
Comment (hanzei) [inline i18n/en.json]: Why is this included here? I thought this is just an arbitrary string. I suppose this should be in the webapp instead.

File: plugin/hooks.go [hunk 3]
```diff
@@ -36,6 +36,12 @@ const (
 	TotalHooksId            = iota
 )
 
+// Exported error string(s) to be used as signal mechanisms between plugins and the server.
+const (
+	// to be returned by MessageWillBePosted to signal a pending post should be dismissed.
```
Comment (lieut-data) [inline plugin/hooks.go]: Can you adopt go-doc here, and write this as:
```
DismissPostError dismisses a pending post when the error is returned from MessageWillBePosted.
```

File: plugin/hooks.go [hunk 4]
```diff
@@ -36,6 +36,12 @@ const (
 	TotalHooksId            = iota
 )
 
+// Exported error string(s) to be used as signal mechanisms between plugins and the server.
```
Comment (lieut-data) [inline plugin/hooks.go]: nit: I'd just remove this comment and let each one stand alone.

**Comment by carmo-evan** [comment]
> Just a minor change request re: GoDoc, but otherwise looks good!

Ready for a second look.

File: plugin/hooks.go [hunk 5]
```diff
@@ -36,6 +36,11 @@ const (
 	TotalHooksId            = iota
 )
 
+const (
+	//DismissPostError dismisses a pending post when the error is returned from MessageWillBePosted.
```
Comment (hanzei) [inline plugin/hooks.go]: Nit: There should be a whitespace.
```suggestion
	// DismissPostError dismisses a pending post when the error is returned from MessageWillBePosted.
```

**Comment by lieut-data** [comment]
Looks good to me @carmo-evan -- did we want to wait for the webapp changes that go along with this before merging?

**Comment by carmo-evan** [comment]
> Looks good to me @carmo-evan -- did we want to wait for the webapp changes that go along with this before merging?

I think it makes sense to wait since this won't really work without it. Would you mind taking a look at the discussion [here](https://github.com/mattermost/mattermost-server/issues/10883)? I have a few comments but Joram is on vacation.

**Comment by cpoile** [comment]
@hanzei can this be merged now? (to support: https://github.com/mattermost/mattermost-redux/pull/844)

**Comment by hanzei** [comment]
Yes!
